app/app.go

```diff
@@ -758,6 +758,11 @@ func (app *App) Name() string { return app.BaseApp.Name() }
 
 // BeginBlocker application updates every begin block
 func (app *App) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+	// enable cross-chain functions for op bnb
+	if ctx.IsUpgraded(upgradetypes.Pampas) {
+		app.CrossChainKeeper.SetDestOpChainID(sdk.ChainID(app.appConfig.CrossChain.DestOpChainId))
```

Comment:
I think this can move to `UpgradeInitializer`, it no needs to be set in every blocks

Reply:
it should be set in every block, it's a local variable.